src/service/xhr-impl.js

```diff
@@ -419,35 +420,31 @@ function isRetriable(status) {
 /**
  * Returns the response if successful or otherwise throws an error.
  * @param {!FetchResponse} response
- * @return {!Promise<!FetchResponse>}
+ * @return {!FetchResponse|!Promise<!FetchResponse>}
```

Comment:
Isn't this a very strange response type? I realize that in a promise chain this doesn't matter, but we should at least rename this method to avoid accidents.